src/main/java/org/spongepowered/api/service/economy/Account.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.service.economy;
+
+import java.math.BigDecimal;
+import java.util.Map;
+import java.util.Set;
+
+import org.spongepowered.api.service.context.Context;
+import org.spongepowered.api.service.context.Contextual;
+import org.spongepowered.api.service.economy.transaction.DepositResult;
+import org.spongepowered.api.service.economy.transaction.ExchangeResult;
+import org.spongepowered.api.service.economy.transaction.TransactionResult;
+import org.spongepowered.api.service.economy.transaction.TransferResult;
+import org.spongepowered.api.service.economy.transaction.WithdrawResult;
+
+/**
+ * TODO: Add overview doc
+ *
+ */
+public interface Account extends Contextual {
+
+    
+    BigDecimal getDefaultBalance(Currency currency);
+
+    /**
+     * Returns a {@link BigDecimal} representative of the balance stored within this
+     * {@link Account} for the {@link Currency} given and the set of {@link Context}s.
+     * 
+     * <p>The default result when the account does not have a balance of the given
+     * {@link Currency} should be {@link BigDecimal#ZERO}.</p>
+     * 
+     * <p>The balance may be unavailable depending on the set of {@link Context}s used.</p>
```

Comment:
If the balance is unavailable, is a runtime exception thrown? 0 returned? null? Maybe it should return an Optional<BigDecimal>